### ISSUE TITLE
[HOLD for payment 2024-06-28] [$250] [MEDIUM] Debugability: Add a keyboard shortcut to open "four finger tap" debug screen on desktop

### ISSUE THREAD

**Issue by muttmuure** [Expensify/App#43256]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## What performance issue do we need to solve?
e.g. memory consumption, storage read/write times, React native bridge concerns, inefficient React component rendering, etc.

When debugging mobile when you are attempting to retrieve client-side logs, you can isolate the logs with the highest context and least noise by using the four-finger tap gesture and enabling at the very moment you need to.

There is no such method of isolating these logs immediately on web and desktop, which means to gather important logs you need to navigate to the Troubleshoot menu and add a bunch of noise and by navigating through the stack add another command which resolves the problematic command you wanted to debug in the first place.

## What is the impact of this on end-users?
List specific user experiences that will be improved by solving this problem e.g. app boot time, time to for some interaction to complete, etc.

This will allow our internal team to debug web issues with the same precision and reliability as mobile.

## List any benchmarks that show the severity of the issue
Please also provide exact steps taken to collect metrics above if any so we can independently verify the results.

N/A

## Proposed solution (if any)
Please list out the steps you think we should take to solve this issue.

Add a keyboard shortcut that opens the debug page like we have on mobile: 

![image](https://github.com/Expensify/App/assets/46995600/179a8c97-e340-477b-85f0-17c90abc2daa)

## List any benchmarks after implementing the changes to show impacts of the proposed solution (if any)
Note: These should be the same as the benchmarks collected before any changes.

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

**Version Number:**
**Reproducible in staging?:**
**Reproducible in production?:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation
**Expensify/Expensify Issue URL:**
**Issue reported by:**
**Slack conversation:**

[View all open jobs on Upwork](https://www.upwork.com/ab/jobs/search/?q=Expensify%20React%20Native&sort=recency&user_location_match=2)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01b24a23b8f2770a3b</li>
        <li>Upwork Job ID: 1800150825662395255</li>
        <li>Last Price Increase: 2024-06-10</li>
<li>Automatic offers: </li>
<ul>
<li>dominictb | Contributor | 102685139</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @jayeshmangwani</details>

**Comment by muttmuure** [comment]
Only client side logging is currently supported on web and desktop natively right now, so the menu options should reflect that.

**Comment by jayeshmangwani** [comment]
Overall @dominictb's  [Proposal](https://github.com/Expensify/App/issues/43256#issuecomment-2157181996) looks good to me. We can move forward with it and Disabling the Client side logging option for platforms other than web and desktop can be discussed in the PR.

@muttmuure What shortcut key should we use for opening the debug screen? CMD + D?

🎀 👀 🎀 C+ reviewed

**Comment by Julesssss** [comment]
CMD + D seems good 👍

**Comment by Julesssss** [comment]
@dominictb as one additional change, could you add the shortcut to the Keyboard shortcut list under `Settings > About`. Thanks

<img width="341" alt="Screenshot 2024-06-11 at 12 58 27" src="https://github.com/Expensify/App/assets/10736861/3be44853-fe6a-4070-a704-a968b8184050">

**Comment by jayeshmangwani** [comment]
> could you add the shortcut to the Keyboard shortcut list under Settings > About. Thanks

The new debug shortcut will be displayed on the KeyboardShortcutsPage. As short cut data comes from the CONST.KEYBOARD_SHORTCUTS and we will define the new debug shortcutKey under KEYBOARD_SHORTCUTS

**Comment by Julesssss** [comment]
Oh nice

**Comment by Julesssss** [comment]
Hey @LLPeckham, in this case we need a description for the shortcut that pens the debugging 'dialog' (not sure why we use Dialog):

<img width="479" alt="image (17)" src="https://github.com/Expensify/App/assets/10736861/a04dd1be-440f-4aae-981a-959a2c0ec32b">

**Comment by LLPeckham** [comment]
@Julesssss can you please share a bit more context or propose an initial line? Just not quite sure what is needed here or what this is for/how to best explain it to a user in plain English.

**Comment by Julesssss** [comment]
Yeah, this is just used to describe a keyboard shortcut for power users. 'Open testing preference' is the initial line proposal. Something like 'Open debug/logging dialog' would be better in my opinion as this dialog lets users enable some special testing features. Here's what is show to the user when they use the shortcut: 

<img width="1195" alt="338923517-e8704359-a72a-4640-bf12-f05177faa840" src="https://github.com/Expensify/App/assets/10736861/43b09948-c190-4824-9dae-522b0efdf63e">

**Comment by Julesssss** [comment]
Actually the more I think about it, the more that the original proposal `Open testing preferences` is a good string that matches the other shortcut descriptions. LMK if you have any suggestions though.

**Comment by LLPeckham** [comment]
Got it, that's v helpful, thanks! Yeah we could either do `Open testing preferences` or if you want to include dialog like the other lines have - would it make sense to say, `Open testing preferences dialog` ?

**Comment by Julesssss** [comment]
Thanks. Agree that 'Open testing preferences dialog' aligns better with the other option.

**Comment by jayeshmangwani** [comment]
- [x]  [@jayeshmangwani] The PR that introduced the bug has been identified. Link to the PR: **This is feature enhancement for the Debugability**
- [x]  [@jayeshmangwani] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: **This is feature enhancement for the Debugability**
- [x]  [@jayeshmangwani] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: **This is a feature enhancement for Debugability. A regression test will be enough.**
- [x]  [@jayeshmangwani] Determine if we should create a regression test for this bug. **Yes**
- [x]  [@jayeshmangwani] If we decide to create a regression test for the bug, please propose the regression test steps to ensure the same bug will not reach production again.



## Regression Test Proposal

1. Open web/desktop app and press CTRL/CMD + D
2. Verify that the troubleshooting modal shown up

Do we agree 👍 or 👎

**Comment by kevinksullivan** [comment]
Summary:
- @jayeshmangwani $250 (NewDot)
- @dominictb $250

**Comment by jayeshmangwani** [comment]
Requested $250 on NewDot

**Comment by JmillsExpensify** [comment]
$250 approved for @jayeshmangwani
